lib/cmap/connection.js

```diff
@@ -122,7 +122,7 @@ class Connection extends EventEmitter {
       if (issue.isTimeout) {
         op.cb(
           new MongoNetworkTimeoutError(`connection ${this.id} to ${this.address} timed out`, {
-            beforeHandshake: !!this[kIsMaster]
+            beforeHandshake: this[kIsMaster] == null
```

Comment:
@nbbeeken what were the cases where `!!this[kIsMaster]` was yielding an incorrect value? we should try to cover them in the tests